trpcultivate_phenotypes/src/TripalCultivateValidator/ValidatorTraits/Headers.php

```diff
@@ -0,0 +1,201 @@
+<?php
+
+namespace Drupal\trpcultivate_phenotypes\TripalCultivateValidator\ValidatorTraits;
+
+/**
+ * Provides setters focused for setting headers used by the importer, 
+ * and getter to retrieve the set value.
+ */
+trait Headers {
+  /**
+   * The key used by the setter method to create a headers element 
+   * in the context array, as well as the key used by the getter method 
+   * to reference and retrieve the headers element values. 
+   * 
+   * Each value is keyed by the types defined by type property.
+   *
+   * @var string
+   */ 
+  private string $context_key = 'headers';
+  
+  /**
+   * Expected types of the headers, each type will have a getter method.
+   * A general-purpose getter getHeaders() can be used to get headers of 
+   * a specific one header type or combination of header types.
+   * 
+   * @var array
+   */
+  private array $types = [
+    'required', // Header must have a value.
+    'optional', // Header may contain a value.
+  ];
+
+  /**
+   * Sets the headers.
+   * 
+   * @param $headers
+   *   The key-value pair array of headers defined by the importer.
+   *   Each item consists of a header name, text description and type (required, optional)
+   *   keyed by name, description and type, respectively.
```

Comment:
No mention of description ;-p

```suggestion
   *   A list of header definitions defined by the importer (usually in a protected
   *   variable at the top of the importer class). Each item in this list is an associative
   *   array defining the specific header and the headers must be listed in order!
   *   Each header item must consist of a header name (key: 'name'), and type (key: 'type',
   *   supported values: 'required', 'optional').
```